http_client.rst

```diff
@@ -1378,12 +1378,25 @@ This allows using them where native PHP streams are needed::
     // later on if you need to, you can access the response from the stream
     $response = stream_get_meta_data($streamResource)['wrapper_data']->getResponse();
 
-Testing HTTP Clients and Responses
-----------------------------------
+Testing
+-------
 
 This component includes the ``MockHttpClient`` and ``MockResponse`` classes to
-use them in tests that need an HTTP client which doesn't make actual HTTP
-requests.
+use them in tests that shouldn't make actual HTTP requests. Such tests can be
+useful, as they are faster and consistent due to not being dependant on
+external service the request would be sent to. In result there is no need to
+worry about the service being online or the request changing some state,
+for example deleting a resource.
+
+``MockHttpClient`` implements the ``HttpClientInterface``, just like any actual
+HTTP client in this component. You can use the interface to accept real client
+in your code, while replacing it with ``MockHttpClient`` in the test.
```

Comment:
```suggestion
HTTP client in this component. When you typehint with ``HttpClientInterface`` your code
will accept the real client in production, while replacing it with ``MockHttpClient`` in the test.
```

Reply:
As discussed on Slack, `outside tests` is better than `production`, just in case someone confuses it with dev/prod envs.